### PR TITLE
0-width bitstrings

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
## Personnel
- [x] **Owner**: @mbudiu-vmw
- [x] **Supporters**: @jafingerhut  
## Design
- [x] **Document**: See spec changes

## Implementation
- [x] **`p4-spec`**: https://github.com/p4lang/p4-spec/pull/931
- [x] **`p4c`**: https://github.com/p4lang/p4c/pull/2724

## Process
- [x] **LDWG discussed**: March 2021 https://docs.google.com/document/d/1XSgdXeG1UuF1FM_XAqxDrHeN4dHZWBnJPKVS6SnGNwM/edit#heading=h.gfxchmii5z9v
- [x] **LDWG approved**: April 2021 https://docs.google.com/document/d/1XSgdXeG1UuF1FM_XAqxDrHeN4dHZWBnJPKVS6SnGNwM/edit#heading=h.smw0dn8iuclh
- [x] **Merged into `p4-spec`**:
- [x] **Merged into `p4c`**:

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -1981,8 +1981,10 @@ An unsigned integer (which we also call a "bit-string") has an
 arbitrary width, expressed in bits. A bit-string of width `W` is
 declared as: `bit<W>`. `W` must be an expression that evaluates to a
 compile-time known value (see Section [#sec-ct-constants]) that is a
-positive integer greater than 0.  When using an expression for the
-size they must be parenthesized.
+positive integer greater or equal to 0.  When using an expression for
+the size they must be parenthesized.  Bitstrings with width 0 are
```
Comment (blp) [inline p4-16/spec/P4-16-spec.mdk:1985]: "expression" is singular, so "they" should be "it".

**Comment by mbudiu-vmw** [comment]
Implement this check in the compiler.

**Comment by mbudiu-vmw** [comment]
I realized that `varbit<0>` should also be supported.

**Comment by mbudiu-vmw** [comment]
If I can get a review on this PR we can probably merge it as well.

**Review by jnfoster — APPROVED** [review]
Looks great. Thanks!

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -2038,7 +2040,7 @@ values, P4 provides a special bit-string type whose size is set at
 runtime, called a `varbit`.
 
 The type `varbit<W>` denotes a bit-string with a width of at most `W`
-bits, where `W` must be a positive integer that is a compile-time
+bits, where `W` must be a positive or zero integer that is a compile-time
```
[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk]: weird, I was pretty sure I had made this change.

**Pull request merged**